pkg/feature/raw_resources.go

```diff
@@ -40,20 +41,31 @@ func createResources(cli client.Client, objects []*unstructured.Unstructured, me
 
 		name := object.GetName()
 		namespace := object.GetNamespace()
+		managed, exists := object.GetAnnotations()[annotations.ManagedByODHOperator]
 
 		err := cli.Get(context.TODO(), k8stypes.NamespacedName{Name: name, Namespace: namespace}, object.DeepCopy())
-		if err == nil {
-			// object already exists, skip reconcile allowing users to tweak it
-			continue
-		}
-		if !k8serrors.IsNotFound(err) {
+		if err != nil && !k8serrors.IsNotFound(err) {
 			return errors.WithStack(err)
 		}
 
-		err = cli.Create(context.TODO(), object)
-		if err != nil {
-			return errors.WithStack(err)
+		if exists && managed == "true" {
+			// update or create object since we manage it
+			if err == nil {
+				if err := cli.Update(context.TODO(), object); err != nil {
```

Comment:
I wonder if we don't want to make server-side apply here, WDYT @aslakknutsen ?

```golang
		data, err := json.Marshal(object)
		if err != nil {
			return err
		}

		if err := cli.Patch(ctx, found, client.RawPatch(types.ApplyPatchType, data), client.ForceOwnership, client.FieldOwner(owner.GetName())); err != nil {
			return err
		}
```

Reply:
Context:

https://github.com/opendatahub-io/opendatahub-operator/blob/6711ab3f1116dd05a229cc18037c31ea7386adbf/pkg/deploy/deploy.go#L316-L322